packages/cli/src/lib/setup/setupUpload.ts

```diff
@@ -753,29 +825,34 @@ function Upload(options) {
                 ioPack.common.installedFrom = ioPackFile.common.installedFrom;
             }
             // Not existing? Why ever ... we recreate
-            let obj;
+            let _obj;
             try {
-                obj = await objects.getObject('system.adapter.' + name);
+                _obj = await this.objects.getObject(`system.adapter.${name}`);
             } catch {
                 // ignore err
             }
-            obj = obj || {};
+            const obj: Partial<ioBroker.AdapterObject> = _obj || {};
             obj.common = ioPack.common || {};
             obj.native = ioPack.native || {};
             // protected/encryptedNative and notifications also need to be updated
             obj.protectedNative = ioPack.protectedNative || [];
             obj.encryptedNative = ioPack.encryptedNative || [];
+            // @ts-expect-error TODO needs to be added to types
             obj.notifications = ioPack.notifications || [];
             // update instanceObjects and objects
+            // @ts-expect-error TODO needs to be added to types
             obj.instanceObjects = ioPack.instanceObjects || [];
+            // @ts-expect-error TODO needs to be added to types
             obj.objects = ioPack.objects || [];
```

Comment:
The goal is to have a full `ioBroker.AdapterObject` at the end. TS can only help you here if you define that from the beginning, so:
```ts
const obj: ioBroker.AdapterObject = {
    common: ioPack.common || {},
    native: ioPack.native || {},
    // ...
};
```
and leave out the `as ioBroker.AdapterObject` at the end.

Reply:
Yes, I thought about it. It will change the logic here. Currently he gets the object and even if the object exists and some properties are missing, it will enhance with io-pack content. If we change it the proposed way, it will not replace on existing objects. Or we let the code be a bit duplicate, so change definition but also hold replace logic.